src/components/webview/SourceControlWebviewPanel.ts

```diff
@@ -5,36 +5,35 @@ import {
 	ViewColumn,
 	Disposable,
 	commands,
-	Webview,
 } from 'vscode';
 import { Message, MessageBus, MessageKind } from '../messageBus';
 import { WebviewResolver } from './WebviewResolver';
 import { View, WebviewMessage, WebviewResponse } from './webviewEvents';
 import { RepositoryService } from './repository';
 import { UserAccountStorage } from '../user/userService';
 
-export class IntuitaPanel {
-	private __view: Webview | null = null;
+export class SourceControlWebviewPanel {
 	private __panel: WebviewPanel | null = null;
 	private __disposables: Disposable[] = [];
-	static __instance: IntuitaPanel | null = null;
+	private __webviewMounted = false;
+	static instance: SourceControlWebviewPanel | null = null;
 
 	static getInstance(
 		context: ExtensionContext,
 		messageBus: MessageBus,
 		repositoryService: RepositoryService,
 		userAccountStorage: UserAccountStorage,
 	) {
-		if (this.__instance) {
-			return this.__instance;
+		if (!SourceControlWebviewPanel.instance) {
```

Comment:
is it possible we don't have a static property at all? so we basically end up with a proper dependency injection mechanism?

Reply:
this will be complicated. panel should follow singleton pattern and should be recreated only if it was disposed. otherwise we will see multiple panels.